packages/browser/src/utils/featureFlags.ts

```diff
@@ -0,0 +1,56 @@
+import type { FeatureFlag } from '@sentry/types';
+
+/**
+ * Ordered LRU cache for storing feature flags in the scope context. The name
+ * of each flag in the buffer is unique, and the output of getAll() is ordered
+ * from oldest to newest.
+ */
+
+/**
+ * Max size of the LRU flag buffer stored in Sentry scope and event contexts.
+ */
+export const FLAG_BUFFER_SIZE = 100;
+
+/**
+ * Insert into a FeatureFlag array while maintaining ordered LRU properties. Not
+ * thread-safe. After inserting:
+ * - `flags` is sorted in order of recency, with the newest flag at the end.
+ * - No other flags with the same name exist in `flags`.
+ * - The length of `flags` does not exceed `maxSize`. The oldest flag is evicted
+ *  as needed.
+ *
+ * @param flags    The array to insert into.
+ * @param name     Name of the feature flag to insert.
+ * @param value    Value of the feature flag.
+ * @param maxSize  Max number of flags the buffer should store. It's recommended
+ *   to keep this consistent across insertions. Default is DEFAULT_MAX_SIZE
+ */
+export function insertToFlagBuffer(
+  flags: FeatureFlag[],
+  name: string,
+  value: boolean,
+  maxSize: number = FLAG_BUFFER_SIZE,
+): void {
+  if (flags.length > maxSize) {
+    throw Error(`insertToFlagBuffer called on a buffer larger than the given maxSize=${maxSize}`);
```

Comment:
m: since we do not catch this I would rather just log this instead of throwing

```suggestion
DEBUG_BUILD && logger.error(`insertToFlagBuffer called on a buffer larger than the given maxSize=${maxSize}`);
```